packages/dd-trace/src/appsec/iast/taint-tracking/taint-tracking-impl.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const TaintedUtils = require('@datadog/native-iast-taint-tracking')
+const { storage } = require('../../../../../datadog-core')
+const iastContextFunctions = require('../iast-context')
+const log = require('../../../log')
+
+function noop (res) { return res }
+const TaintTrackingDummy = {
+  plusOperator: noop,
+  trim: noop,
+  trimEnd: noop
+}
+
+function getTransactionId () {
+  const store = storage.getStore()
+  const iastContext = iastContextFunctions.getIastContext(store)
+  return iastContext && iastContext[iastContextFunctions.IAST_TRANSACTION_ID]
+}
+
+function getFilteredCsiFn (cb, filter) {
+  return function csiCall (res, fn, target, ...rest) {
+    try {
+      if (filter(res, fn, target)) { return res }
+      const transactionId = getTransactionId()
+      if (transactionId) {
+        return cb(transactionId, res, target, ...rest)
+      }
+    } catch (e) {
+      log.debug(e)
+    }
+    return res
+  }
+}
+
+function notString () {
+  return Array.prototype.some.call(arguments, (p) => typeof p !== 'string')
+}
+
+function isValidCsiMethod (fn, ...protos) {
+  return protos.some(proto => fn === proto)
+}
+
+function getCsiFn (cb, ...protos) {
+  let filter
+  if (!protos || protos.length === 0) {
+    filter = (res, fn, target) => notString(res, target)
+  } else if (protos.length === 1) {
+    const protoFn = protos[0]
+    filter = (res, fn, target) => notString(res, target) || fn !== protoFn
+  } else {
+    filter = (res, fn, target) => notString(res, target) || !isValidCsiMethod(fn, ...protos)
```

Comment:
you are converting array to params, to convert params to array in line 40.
Why not something like: `!isValidCsiMethod(fn, protos)`
And change line 40 to:
```
function isValidCsiMethod (fn, ...protos) {
```